sdcflows/interfaces/bspline.py

```diff
@@ -124,27 +131,50 @@ class BSplineApprox(SimpleInterface):
 
     def _run_interface(self, runtime):
         from sklearn import linear_model as lm
-        from scipy.sparse import vstack as sparse_vstack
+
+        # Output name baseline
+        out_name = fname_presuffix(
+            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        )
 
         # Load in the fieldmap
         fmapnii = nb.load(self.inputs.in_data)
-        data = fmapnii.get_fdata(dtype="float32")
+        zooms = fmapnii.header.get_zooms()
 
-        # Generate the output naming base
-        out_name = fname_presuffix(
-            self.inputs.in_data, suffix="_field", newpath=runtime.cwd
+        # Get a mask (or define on the spot to cover the full extent)
+        masknii = (
+            nb.load(self.inputs.in_mask)
+            if isdefined(self.inputs.in_mask)
+            else None
         )
-        # Create a copy of the header for use below
-        hdr = fmapnii.header.copy()
-        hdr.set_data_dtype("float32")
 
+        need_resize = np.any(np.array(zooms) < self.inputs.zooms_min)
+        if need_resize:
+            from sdcflows.utils.tools import resample_to_zooms
+
+            zooms_min = self.inputs.zooms_min
+
+            with suppress(TypeError):
+                zooms_min = [float(zooms_min)] * 3
+
+            LOGGER.info(
+                "Resampling image with resolution exceeding 'zooms_min' "
+                f"({'x'.join(str(s) for s in zooms)})."
+            )
+            fmapnii = resample_to_zooms(fmapnii, zooms_min)
+
+            if masknii is not None:
+                masknii = resample_to_zooms(masknii, zooms_min)
+
+        data = fmapnii.get_fdata(dtype="float32")
+
+        # Generate a numpy array with the mask
         mask = (
-            nb.load(self.inputs.in_mask).get_fdata() > 0
-            if isdefined(self.inputs.in_mask)
-            else np.ones_like(data, dtype=bool)
+            np.ones_like(fmapnii.dataobj, dtype=bool) if masknii is None
```

Comment:
Not sure if numpy would try to coerce to an array, but we don't need this.

```suggestion
            np.ones(fmapnii.shape, dtype=bool) if masknii is None
```